src/swarm-lang/Swarm/Language/Parser/Term.hs

```diff
@@ -109,6 +114,21 @@ sLet x ty t1 = SLet (lvVar x `S.member` setOf freeVarsV t1) x ty t1
 sDef :: LocVar -> Maybe Polytype -> Syntax -> Term
 sDef x ty t = SDef (lvVar x `S.member` setOf freeVarsV t) x ty t
 
+-- | Create a polytype from a list of variable binders and a type.
+--   Ensure that no binder is repeated, and all type variables in the
+--   type are present in the list of binders (/i.e./ the type contains
+--   no free type variables).
+bindTydef :: [Var] -> Type -> Parser Polytype
+bindTydef xs ty
+  | Just repeated <- findDup xs = fail $ "Duplicate variable on left-hand side of tydef: " ++ into @String repeated
+  | not (S.null free) =
+      fail $
```

Comment:
Maybe we want to create a `failWords` utility method that has the convenience of builtin `unwords` like `failT`, but yields a string output.

Reply:
Good idea!  It turns out we can just use `failT` itself here, since it already works in any `MonadFail`, which I had forgotten.